modules/ROOT/pages/salesforce/salesforce-connector-97.adoc

```diff
@@ -0,0 +1,1274 @@
+   	fetchAllApexSoapMetadata="true" 
+	fetchAllApexRestMetadata="true">
+	<salesforce:apex-class-names >
+	     <salesforce:apex-class-name value="xyz" />
+	</salesforce:apex-class-names>
+</salesforce:sfdc-config>
+----
+
+== Create a Consumer Key
+
+A consumer key is required when setting up OAuth 2.0 configurations for the Salesforce connector. It is used by the OAuth JWT and SAML bearer configurations and by the OAuth Username Password configuration.
+
+This procedure provides guidance on using Salesforce to create a consumer key, and explains how to create a connected app in Salesforce. However, note that the steps might differ somewhat in your Salesforce instance.
+
+Prerequisite:
+
+This procedure assumes that you already have a certification file (such as `salesforce-cert.crt`). If not, you can produce one by generating a Java KeyStore and Public Key.
+
+[[create-consumer-key]]
+. Log into Salesforce, and go to *Setup* > *Build* > *Create* > *Apps*.
+. Under the Connected App section, click *New*.
+. Follow these steps to create a new connected app, and enter the following information in the appropriate fields:
++
+* A name for the connected app.
+* The API name.
+* Contact email.
++
+. Under API (Enable OAuth Settings), select *Enable OAuth Settings*:
++
+* Enter the *Callback URL*.
+* Select the *Use Digital Signatures* checkbox.
+* Click *Browse* (or *Choose File*), and load your Salesforce certificate (for example, `salesforce-cert.crt`), which contains your public key.
++
+In Studio, you typically store this in the workspace that contains your Mule application.
++
+. Add and Save these OAuth scopes to Selected OAuth Scopes:
++
+*Full Access* (`full`) and *Perform Requests On Your Behalf At Any Time* (`refresh_token`, `offline_access`)
++
+. Configure the Authorization Settings for the app.
+. Click *Manage*. Then under the OAuth Policies section, expand the *Permitted Users* dropdown, and select *Admin Approved Users are Pre-Authorized*. Then *Save*.
+. Under the Profiles section, click *Manage Profiles*.
+. Select your user profile, and then click *Save*.
+. Go back to the list of Connected Apps: *Build* > *Create* > *Apps*.
+. Under the Connected Apps section, select the connected app you created.
+
+You can see the Consumer Key that you need to provide in your connector's configuration.
+
+== Generate a Keystore File
+
+The Keystore is the path to the keystore used to sign data during authentication. Only the Java keystore (JKS) format is allowed.
+
+To generate a keystore file:
+
+. Go to your Mule workspace, and open the command prompt (for Windows) or Terminal (for Mac).
+. Type this command and press enter:
++
+[source]
+----
+keytool -genkeypair -alias salesforce-cert -keyalg RSA -keystore salesforce-cert.jks
+----
++
+. Enter the following:
++
+** Password for the keystore.
+** Your first name and last name.
+** Your organization unit.
+** Name of your city, state, and the two letters code of your county.

```

Comment:
Will this be true for the Japan version?

Reply:
Good catch -- county was a typo for country.